Dockerfile

```diff
@@ -9,15 +9,16 @@ MAINTAINER Lars Gierth <lgierth@ipfs.io>
 ENV GX_IPFS ""
 ENV SRC_DIR /go/src/github.com/ipfs/go-ipfs
 
-COPY . $SRC_DIR
+COPY ./package.json $SRC_DIR/package.json
 
-# Build the thing.
-# Also: fix getting HEAD commit hash via git rev-parse.
+# Fetch dependencies.
 # Also: allow using a custom IPFS API endpoint.
-RUN cd $SRC_DIR \
-  && mkdir .git/objects \
+RUN set -x \
+  && go get github.com/whyrusleeping/gx \
```

Comment:
I would prefer if `gx` and `gx-go` were copied over from the `bin/` directory instead of getting the latest version every time so that way you have one less possible variation between a local build env and the docker daemon build env.

Reply:
Actually, I'm pretty sure we neither want nor need gx in the docker build. We only need to do this here because we haven't run `make build` yet (which installs gx). Instead of this, I believe we should be running `make deps` (Dockerfile.fast should be doing the same thing).

Reply:
`make deps` just install `gx,gx-go` if they aren't present which will be the case since the whole directory isn't being copied in. The added issue is that `make deps` uses `dist_get` to install `gx,gx-go` and `dist_get` also is missing due to the `bin/` directory not being copied into the docker build context. Though using `make deps` solves the which version of `gx,gx-go` is being grabbed issue. We still need to copy in `bin/dist_get` at the least though to make it work.

Reply:
I see. Yes, we should be copying bin (I guess? I haven't read the docker documentation)